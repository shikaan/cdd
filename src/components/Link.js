import React from 'react'
import {Link as GatsbyLink, navigate as gastbyNavigate} from 'gatsby'
import styled from 'styled-components'

const UnstyledButton = styled.button`
  background:none !important;
  border:none; 
  padding:0!important;
  cursor: pointer;
  text-align: unset;
  
  &:focus {outline:0;}
`

const UnstyledLink = ({href, onClick, ...rest}) => {
  if (href) {
    return React.createElement('a', {href, ...rest})
  } else if (onClick) {
    return React.createElement(UnstyledButton, {onClick, ...rest})
  }

  return React.createElement(GatsbyLink, rest)
}

export const navigate = gastbyNavigate

const Link = styled(UnstyledLink)(({theme}) => `
  font-family: ${theme.typography.primaryFont.fontFamily};
  font-weight: 700;
  text-decoration: none;
`)

export default Link
