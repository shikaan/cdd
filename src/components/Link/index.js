import React from 'react'
import { Link as GatsbyLink } from 'gatsby'
import styled from 'styled-components'

const UnstyledButton = styled.button`
  background:none !important;
  border:none; 
  padding:0!important;
  cursor: pointer;
  
  &:focus {outline:0;}
`

const Link = ({ href, onClick, children, ...rest }) => {
  if (href) {
    return React.createElement('a', { children, href, ...rest })
  } else if (onClick) {
    return React.createElement(UnstyledButton, { children, onClick, ...rest })
  }

  return React.createElement(GatsbyLink, { children, ...rest })
}

export default styled(Link)`
  text-decoration: none;
`
