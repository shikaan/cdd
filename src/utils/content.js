/**
 * @param {Array<{reference: string}>} microcopy
 * @param {string} key
 * @return {string}
 */
export const getMicrocopy = (microcopy, key) => {
  const element = microcopy?.find(i => i.reference === key) ?? {};

  return element.value ?? "";
};

/**
 * @param {Array<{reference: string}>} sections
 * @param {string} key
 * @return {string}
 */
export const getSection = (sections, key) => {
  return sections?.find(i => i.reference === key) ?? {reference: key};
};