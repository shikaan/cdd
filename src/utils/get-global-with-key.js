export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};
