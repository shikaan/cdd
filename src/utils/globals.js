export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};

export const getGlobal = () => { 
  // eslint-disable-next-line no-restricted-globals
  if (typeof self !== "undefined") { return self }
  if (typeof window !== "undefined") { return window } 
  if (typeof global !== "undefined") { return global } 
  throw new Error("unable to locate global object"); 
}; 
