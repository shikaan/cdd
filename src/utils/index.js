export {isLastIndex} from "./is-last-index";
