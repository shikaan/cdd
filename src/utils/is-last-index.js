export const isLastIndex = (list, index) => {
  if (!Array.isArray(list)) {
    return false;
  }

  return index === list.length - 1;
};
