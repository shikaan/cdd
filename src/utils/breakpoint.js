import {numberBreakpoint} from "~theme/variables";
import {getGlobalWithKey} from "~/utils/get-global-with-key";

export const getViewportWidth = () => getGlobalWithKey("innerWidth") ?? getGlobalWithKey("document").documentElement.clientWidth;

export const isMobile = () => {
  const viewportWidth = getViewportWidth();
  return viewportWidth < numberBreakpoint.sm;
};

export const isTablet = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.sm <= viewportWidth && viewportWidth < numberBreakpoint.md;
};

export const isDesktop = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.md <= viewportWidth;
};