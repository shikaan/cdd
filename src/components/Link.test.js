import React from "react";
import {shallow, mount} from "enzyme";
import "jest-styled-components";
import {noop} from "lodash";

import {getTagNameByHTMLString, mountWithTheme} from "/test/utils";
import themeMock from "~theme";

import Link from "./Link";
import {Link as GatsbyLink} from "gatsby";

describe("Link", () => {
  it("renders a button if onClick is set", () => {
    const wrapper = shallow(<Link theme={themeMock} onClick={noop}>link</Link>);

    expect(getTagNameByHTMLString(wrapper.html())).toEqual("button");
  });

  it("renders an anchor if href is passed", () => {
    const wrapper = shallow(<Link theme={themeMock} href={"somewhere"}>link</Link>);

    expect(getTagNameByHTMLString(wrapper.html())).toEqual("a");
  });

  it("renders Gatsby Link else", () => {
    const wrapper = mountWithTheme(<Link to={"somewhere-else"}>link</Link>, themeMock);

    expect(wrapper.childAt(0).childAt(0).childAt(0).type()).toEqual(GatsbyLink);
  });

  it("renders children", () => {
    const children = "text";
    const wrapper = mount(<Link theme={themeMock} to={"somewhere-else"} children={children}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
