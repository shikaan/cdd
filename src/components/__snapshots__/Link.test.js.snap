// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders children 1`] = `
.c0 {
  font-family: Merriweather,Georgia,serif;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Styled(UnstyledLink)
  theme={
    Object {
      "breakpoint": Object {
        "lg": "1184px",
        "md": "980px",
        "mg": "1920px",
        "sm": "768px",
        "xl": "1366px",
        "xs": "480px",
      },
      "color": Object {
        "brand100": "hsla(23,92%,95%,1)",
        "brand200": "hsla(23,92%,82%,1)",
        "brand300": "hsla(23,92%,67%,1)",
        "brand400": "hsla(23,92%,52%,1)",
        "brand500": "hsla(23,92%,35%,1)",
        "brand600": "hsla(23,92%,18%,1)",
        "dark100": "hsla(44,8%,95%,1)",
        "dark200": "hsla(44,8%,82%,1)",
        "dark300": "hsla(44,8%,67%,1)",
        "dark400": "hsla(44,8%,52%,1)",
        "dark500": "hsla(44,8%,35%,1)",
        "dark600": "hsla(44,8%,18%,1)",
        "light": "hsla(0,100%,100%, 1)",
        "withTransparency": Object {
          "brand100": "hsla(23,92%,95%,.5)",
          "brand200": "hsla(23,92%,82%,.5)",
          "brand300": "hsla(23,92%,67%,.5)",
          "brand400": "hsla(23,92%,52%,.5)",
          "brand500": "hsla(23,92%,35%,.5)",
          "brand600": "hsla(23,92%,18%,.5)",
          "dark100": "hsla(44,8%,95%,.5)",
          "dark200": "hsla(44,8%,82%,.5)",
          "dark300": "hsla(44,8%,67%,.5)",
          "dark400": "hsla(44,8%,52%,.5)",
          "dark500": "hsla(44,8%,35%,.5)",
          "dark600": "hsla(44,8%,18%,.5)",
          "light": "hsla(0,100%,100%, .5)",
        },
      },
      "typography": Object {
        "__esModule": true,
        "baseFontSize": Size {
          "_rank": 2,
        },
        "baseLineHeight": "normal",
        "primaryFont": Object {
          "fontFamily": "Merriweather, Georgia, serif",
          "hover": "hsla(44,8%,18%,1)",
        },
        "secondaryFont": Object {
          "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
          "hover": "hsla(44,8%,18%,1)",
        },
      },
      "zIndexRank": Object {
        "0": "0",
        "1": "100",
        "2": "200",
        "3": "300",
        "4": "400",
        "5": "500",
        "6": "600",
        "7": "700",
        "8": "800",
        "9": "900",
        "bottom": "-100",
        "top": "1000",
      },
    }
  }
  to="somewhere-else"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            [Function],
          ],
        },
        "displayName": "Styled(UnstyledLink)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "breakpoint": Object {
          "lg": "1184px",
          "md": "980px",
          "mg": "1920px",
          "sm": "768px",
          "xl": "1366px",
          "xs": "480px",
        },
        "color": Object {
          "brand100": "hsla(23,92%,95%,1)",
          "brand200": "hsla(23,92%,82%,1)",
          "brand300": "hsla(23,92%,67%,1)",
          "brand400": "hsla(23,92%,52%,1)",
          "brand500": "hsla(23,92%,35%,1)",
          "brand600": "hsla(23,92%,18%,1)",
          "dark100": "hsla(44,8%,95%,1)",
          "dark200": "hsla(44,8%,82%,1)",
          "dark300": "hsla(44,8%,67%,1)",
          "dark400": "hsla(44,8%,52%,1)",
          "dark500": "hsla(44,8%,35%,1)",
          "dark600": "hsla(44,8%,18%,1)",
          "light": "hsla(0,100%,100%, 1)",
          "withTransparency": Object {
            "brand100": "hsla(23,92%,95%,.5)",
            "brand200": "hsla(23,92%,82%,.5)",
            "brand300": "hsla(23,92%,67%,.5)",
            "brand400": "hsla(23,92%,52%,.5)",
            "brand500": "hsla(23,92%,35%,.5)",
            "brand600": "hsla(23,92%,18%,.5)",
            "dark100": "hsla(44,8%,95%,.5)",
            "dark200": "hsla(44,8%,82%,.5)",
            "dark300": "hsla(44,8%,67%,.5)",
            "dark400": "hsla(44,8%,52%,.5)",
            "dark500": "hsla(44,8%,35%,.5)",
            "dark600": "hsla(44,8%,18%,.5)",
            "light": "hsla(0,100%,100%, .5)",
          },
        },
        "typography": Object {
          "__esModule": true,
          "baseFontSize": Size {
            "_rank": 2,
          },
          "baseLineHeight": "normal",
          "primaryFont": Object {
            "fontFamily": "Merriweather, Georgia, serif",
            "hover": "hsla(44,8%,18%,1)",
          },
          "secondaryFont": Object {
            "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
            "hover": "hsla(44,8%,18%,1)",
          },
        },
        "zIndexRank": Object {
          "0": "0",
          "1": "100",
          "2": "200",
          "3": "300",
          "4": "400",
          "5": "500",
          "6": "600",
          "7": "700",
          "8": "800",
          "9": "900",
          "bottom": "-100",
          "top": "1000",
        },
      }
    }
    to="somewhere-else"
  >
    <UnstyledLink
      className="c0"
      theme={
        Object {
          "breakpoint": Object {
            "lg": "1184px",
            "md": "980px",
            "mg": "1920px",
            "sm": "768px",
            "xl": "1366px",
            "xs": "480px",
          },
          "color": Object {
            "brand100": "hsla(23,92%,95%,1)",
            "brand200": "hsla(23,92%,82%,1)",
            "brand300": "hsla(23,92%,67%,1)",
            "brand400": "hsla(23,92%,52%,1)",
            "brand500": "hsla(23,92%,35%,1)",
            "brand600": "hsla(23,92%,18%,1)",
            "dark100": "hsla(44,8%,95%,1)",
            "dark200": "hsla(44,8%,82%,1)",
            "dark300": "hsla(44,8%,67%,1)",
            "dark400": "hsla(44,8%,52%,1)",
            "dark500": "hsla(44,8%,35%,1)",
            "dark600": "hsla(44,8%,18%,1)",
            "light": "hsla(0,100%,100%, 1)",
            "withTransparency": Object {
              "brand100": "hsla(23,92%,95%,.5)",
              "brand200": "hsla(23,92%,82%,.5)",
              "brand300": "hsla(23,92%,67%,.5)",
              "brand400": "hsla(23,92%,52%,.5)",
              "brand500": "hsla(23,92%,35%,.5)",
              "brand600": "hsla(23,92%,18%,.5)",
              "dark100": "hsla(44,8%,95%,.5)",
              "dark200": "hsla(44,8%,82%,.5)",
              "dark300": "hsla(44,8%,67%,.5)",
              "dark400": "hsla(44,8%,52%,.5)",
              "dark500": "hsla(44,8%,35%,.5)",
              "dark600": "hsla(44,8%,18%,.5)",
              "light": "hsla(0,100%,100%, .5)",
            },
          },
          "typography": Object {
            "__esModule": true,
            "baseFontSize": Size {
              "_rank": 2,
            },
            "baseLineHeight": "normal",
            "primaryFont": Object {
              "fontFamily": "Merriweather, Georgia, serif",
              "hover": "hsla(44,8%,18%,1)",
            },
            "secondaryFont": Object {
              "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
              "hover": "hsla(44,8%,18%,1)",
            },
          },
          "zIndexRank": Object {
            "0": "0",
            "1": "100",
            "2": "200",
            "3": "300",
            "4": "400",
            "5": "500",
            "6": "600",
            "7": "700",
            "8": "800",
            "9": "900",
            "bottom": "-100",
            "top": "1000",
          },
        }
      }
      to="somewhere-else"
    >
      <mockConstructor
        className="c0"
        theme={
          Object {
            "breakpoint": Object {
              "lg": "1184px",
              "md": "980px",
              "mg": "1920px",
              "sm": "768px",
              "xl": "1366px",
              "xs": "480px",
            },
            "color": Object {
              "brand100": "hsla(23,92%,95%,1)",
              "brand200": "hsla(23,92%,82%,1)",
              "brand300": "hsla(23,92%,67%,1)",
              "brand400": "hsla(23,92%,52%,1)",
              "brand500": "hsla(23,92%,35%,1)",
              "brand600": "hsla(23,92%,18%,1)",
              "dark100": "hsla(44,8%,95%,1)",
              "dark200": "hsla(44,8%,82%,1)",
              "dark300": "hsla(44,8%,67%,1)",
              "dark400": "hsla(44,8%,52%,1)",
              "dark500": "hsla(44,8%,35%,1)",
              "dark600": "hsla(44,8%,18%,1)",
              "light": "hsla(0,100%,100%, 1)",
              "withTransparency": Object {
                "brand100": "hsla(23,92%,95%,.5)",
                "brand200": "hsla(23,92%,82%,.5)",
                "brand300": "hsla(23,92%,67%,.5)",
                "brand400": "hsla(23,92%,52%,.5)",
                "brand500": "hsla(23,92%,35%,.5)",
                "brand600": "hsla(23,92%,18%,.5)",
                "dark100": "hsla(44,8%,95%,.5)",
                "dark200": "hsla(44,8%,82%,.5)",
                "dark300": "hsla(44,8%,67%,.5)",
                "dark400": "hsla(44,8%,52%,.5)",
                "dark500": "hsla(44,8%,35%,.5)",
                "dark600": "hsla(44,8%,18%,.5)",
                "light": "hsla(0,100%,100%, .5)",
              },
            },
            "typography": Object {
              "__esModule": true,
              "baseFontSize": Size {
                "_rank": 2,
              },
              "baseLineHeight": "normal",
              "primaryFont": Object {
                "fontFamily": "Merriweather, Georgia, serif",
                "hover": "hsla(44,8%,18%,1)",
              },
              "secondaryFont": Object {
                "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
                "hover": "hsla(44,8%,18%,1)",
              },
            },
            "zIndexRank": Object {
              "0": "0",
              "1": "100",
              "2": "200",
              "3": "300",
              "4": "400",
              "5": "500",
              "6": "600",
              "7": "700",
              "8": "800",
              "9": "900",
              "bottom": "-100",
              "top": "1000",
            },
          }
        }
        to="somewhere-else"
      >
        <a
          className="c0"
          href="somewhere-else"
          theme={
            Object {
              "breakpoint": Object {
                "lg": "1184px",
                "md": "980px",
                "mg": "1920px",
                "sm": "768px",
                "xl": "1366px",
                "xs": "480px",
              },
              "color": Object {
                "brand100": "hsla(23,92%,95%,1)",
                "brand200": "hsla(23,92%,82%,1)",
                "brand300": "hsla(23,92%,67%,1)",
                "brand400": "hsla(23,92%,52%,1)",
                "brand500": "hsla(23,92%,35%,1)",
                "brand600": "hsla(23,92%,18%,1)",
                "dark100": "hsla(44,8%,95%,1)",
                "dark200": "hsla(44,8%,82%,1)",
                "dark300": "hsla(44,8%,67%,1)",
                "dark400": "hsla(44,8%,52%,1)",
                "dark500": "hsla(44,8%,35%,1)",
                "dark600": "hsla(44,8%,18%,1)",
                "light": "hsla(0,100%,100%, 1)",
                "withTransparency": Object {
                  "brand100": "hsla(23,92%,95%,.5)",
                  "brand200": "hsla(23,92%,82%,.5)",
                  "brand300": "hsla(23,92%,67%,.5)",
                  "brand400": "hsla(23,92%,52%,.5)",
                  "brand500": "hsla(23,92%,35%,.5)",
                  "brand600": "hsla(23,92%,18%,.5)",
                  "dark100": "hsla(44,8%,95%,.5)",
                  "dark200": "hsla(44,8%,82%,.5)",
                  "dark300": "hsla(44,8%,67%,.5)",
                  "dark400": "hsla(44,8%,52%,.5)",
                  "dark500": "hsla(44,8%,35%,.5)",
                  "dark600": "hsla(44,8%,18%,.5)",
                  "light": "hsla(0,100%,100%, .5)",
                },
              },
              "typography": Object {
                "__esModule": true,
                "baseFontSize": Size {
                  "_rank": 2,
                },
                "baseLineHeight": "normal",
                "primaryFont": Object {
                  "fontFamily": "Merriweather, Georgia, serif",
                  "hover": "hsla(44,8%,18%,1)",
                },
                "secondaryFont": Object {
                  "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
                  "hover": "hsla(44,8%,18%,1)",
                },
              },
              "zIndexRank": Object {
                "0": "0",
                "1": "100",
                "2": "200",
                "3": "300",
                "4": "400",
                "5": "500",
                "6": "600",
                "7": "700",
                "8": "800",
                "9": "900",
                "bottom": "-100",
                "top": "1000",
              },
            }
          }
        >
          text
        </a>
      </mockConstructor>
    </UnstyledLink>
  </StyledComponent>
</Styled(UnstyledLink)>
`;
